src/aap_eda/services/project/scm.py

```diff
@@ -0,0 +1,287 @@
+                key_password = inputs.get("ssh_key_unlock")
+                if key_password:
+                    cls.decrypt_key_file(key_file.name, key_password)
+
+        if gpg_credential:
+            gpg_inputs = inputs_from_store(
+                gpg_credential.inputs.get_secret_value()
+            )
+            gpg_key = gpg_inputs.get("gpg_public_key")
+            gpg_key_file = tempfile.NamedTemporaryFile("w+t")
+            gpg_key_file.write(gpg_key)
+            gpg_key_file.flush()
+            extra_vars["verify_commit"] = "true"
+            cls.add_gpg_key(gpg_key_file.name)
+
+        if not verify_ssl:
+            extra_vars["ssl_no_verify"] = "true"
+
+        extra_vars["scm_url"] = final_url
+        if branch:
+            extra_vars["scm_branch"] = branch
+        if refspec:
+            extra_vars["scm_refspec"] = refspec
+        if depth:
+            extra_vars["depth"] = depth
+
+        logger.info("Cloning repository: %s", url)
+        try:
+            with contextlib.chdir(path):
+                git_hash = _executor(extra_vars=extra_vars)
+        except ScmError as e:
+            msg = str(e)
+            if secret:
+                msg = msg.replace(secret, "****", 1)
+            logger.warning("SCM clone failed: %s", msg)
+            raise ScmError(msg) from None
+        finally:
+            if key_file:
+                key_file.close()
+            if gpg_key_file:
+                gpg_key_file.close()
+        instance = cls(path, _executor=_executor)
+        instance.git_hash = git_hash
+        return instance
+
+    @classmethod
+    def build_url(
+        cls, url: str, user: str, password: str, ssh_key: str
+    ) -> str:
+        result = urlparse(url)
+        domain = result.netloc.split("@")[-1]
+        scheme = result.scheme
+        path = result.path
+
+        if ssh_key and scheme == "" and path.startswith("git@"):
+            return url
+        elif user and password:
+            encoded_user = quote(user)
+            encoded_password = quote(password)
+            domain = f"{encoded_user}:{encoded_password}@{domain}"
+        elif password:
+            encoded_token = quote(password)
+            domain = f"{encoded_token}@{domain}"
+
+        unparsed = (
+            scheme,
+            domain,
+            path,
+            result.params,
+            result.query,
+            result.fragment,
+        )
+        return urlunparse(unparsed)
+
+    @classmethod
+    def decrypt_key_file(cls, key_file: str, password: str) -> None:
+        cmd = f'{KEYGEN_COMMAND} -p -P {password} -N "" -f {key_file}'
+        child = pexpect.spawn(cmd)
+        index = child.expect(["Failed to load key", pexpect.EOF])
+        if index == 0:
+            raise ScmError("Incorrect passhprase for the private key")
+
+    @classmethod
+    def add_gpg_key(cls, key_file: str) -> None:
+        cmd = f"{GPG_COMMAND} --import {key_file}"
+        pexpect.run(cmd)
```

Comment:
If the content of the key is wrong this command will fail. Don't we want to handle that case?